File: ios/InciTeam/InciTeam/InciTeamFeature.swift
import SwiftUI

enum InciTeamFeature: String, CaseIterable, Identifiable, Hashable {
    case roster
    case schedule
    case teamMembers
    case leaves
    case breaks
    case configurationItems
    case ciUserMapping
    case summary
    case logs
    case diagnostics
    case accountSettings
    case userAccess

    var id: String {
        rawValue
    }

    var title: String {
        switch self {
        case .roster:
            return "Roster"
        case .schedule:
            return "Schedule"
        case .teamMembers:
            return "Team Members"
        case .leaves:
            return "Leaves"
        case .breaks:
            return "Breaks"
        case .configurationItems:
            return "CI"
        case .ciUserMapping:
            return "CI User Mapping"
        case .summary:
            return "Summary"
        case .logs:
            return "Logs"
        case .diagnostics:
            return "Diagnostics"
        case .accountSettings:
            return "Account"
        case .userAccess:
            return "User Access"
        }
    }

    var subtitle: String {
        switch self {
        case .roster:
            return "On-shift people"
        case .schedule:
            return "Coverage calendar"
        case .teamMembers:
            return "Team directory"
        case .leaves:
            return "Planned absences"
        case .breaks:
            return "Active breaks"
        case .configurationItems:
            return "Supported systems"
        case .ciUserMapping:
            return "Routing ownership"
        case .summary:
            return "Operational health"
        case .logs:
            return "Assignment timeline"
        case .diagnostics:
            return "Routing checks"
        case .accountSettings:
            return "Profile and deletion"
        case .userAccess:
            return "Roles and teams"
        }
    }

    var systemImage: String {
        switch self {
        case .roster:
            return "person.3.sequence.fill"
        case .schedule:
            return "calendar"
        case .teamMembers:
            return "person.2.fill"
        case .leaves:
            return "figure.walk.departure"
        case .breaks:
            return "cup.and.saucer.fill"
        case .configurationItems:
            return "server.rack"
        case .ciUserMapping:
            return "point.3.connected.trianglepath.dotted"
        case .summary:
            return "gauge.with.dots.needle.bottom.50percent"
        case .logs:
            return "list.bullet.rectangle.portrait.fill"
        case .diagnostics:
            return "stethoscope"
        case .accountSettings:
            return "person.crop.circle.badge.exclamationmark"
        case .userAccess:
            return "person.badge.key.fill"
        }
    }

    var tone: FeatureTone {
        switch self {
        case .roster, .schedule, .summary, .logs:
            return .blue
        case .teamMembers, .configurationItems, .ciUserMapping:
            return .slate
        case .leaves:
            return .orange
        case .breaks:
            return .green
        case .diagnostics:
            return .purple
        case .accountSettings, .userAccess:
            return .red
        }
    }
}

struct InciTeamFeatureSection: Identifiable {
    let title: String
    let systemImage: String
    let features: [InciTeamFeature]

    var id: String {
        title
    }

    static let all: [InciTeamFeatureSection] = [
        InciTeamFeatureSection(
            title: "Roster and Schedule",
            systemImage: "calendar.badge.clock",
            features: [.roster, .schedule]
        ),
        InciTeamFeatureSection(
            title: "People",
            systemImage: "person.2.fill",
            features: [.teamMembers]
        ),
        InciTeamFeatureSection(
            title: "Availability",
            systemImage: "person.crop.circle.badge.clock",
            features: [.leaves, .breaks]
        ),
        InciTeamFeatureSection(
            title: "CI Routing",
            systemImage: "arrow.triangle.branch",
            features: [.configurationItems, .ciUserMapping]
        ),
        InciTeamFeatureSection(
            title: "Operations",
            systemImage: "waveform.path.ecg.rectangle",
            features: [.summary, .logs, .diagnostics]
        ),
        InciTeamFeatureSection(
            title: "Access",
            systemImage: "lock.shield.fill",
            features: [.accountSettings, .userAccess]
        )
    ]
}

File: android/app/src/main/java/com/inciteam/app/ui/InciTeamFeature.kt
package com.inciteam.app.ui

import androidx.compose.ui.graphics.Color

enum class FeatureTone(val color: Color) {
    Blue(Color(0xFF1859D1)),
    Green(Color(0xFF1F9D68)),
    Orange(Color(0xFFD48806)),
    Purple(Color(0xFF6750A4)),
    Red(Color(0xFFC74444)),
    Slate(Color(0xFF475569))
}

data class InciTeamFeature(
    val id: String,
    val title: String,
    val subtitle: String,
    val badge: String,
    val tone: FeatureTone
)

data class InciTeamFeatureSection(
    val title: String,
    val badge: String,
    val features: List<InciTeamFeature>
)

val InciTeamFeatureSections = listOf(
    InciTeamFeatureSection(
        title = "Roster and Schedule",
        badge = "RS",
        features = listOf(
            InciTeamFeature(
                id = "roster",
                title = "Roster",
                subtitle = "On-shift people",
                badge = "RO",
                tone = FeatureTone.Blue
            ),
            InciTeamFeature(
                id = "schedule",
                title = "Schedule",
                subtitle = "Coverage calendar",
                badge = "SC",
                tone = FeatureTone.Blue
            )
        )
    ),
    InciTeamFeatureSection(
        title = "People",
        badge = "PE",
        features = listOf(
            InciTeamFeature(
                id = "team-members",
                title = "Team Members",
                subtitle = "Team directory",
                badge = "TM",
                tone = FeatureTone.Slate
            )
        )
    ),
    InciTeamFeatureSection(
        title = "Availability",
        badge = "AV",
        features = listOf(
            InciTeamFeature(
                id = "leaves",
                title = "Leaves",
                subtitle = "Planned absences",
                badge = "LV",
                tone = FeatureTone.Orange
            ),
            InciTeamFeature(
                id = "breaks",
                title = "Breaks",
                subtitle = "Active breaks",
                badge = "BR",
                tone = FeatureTone.Green
            )
        )
    ),
    InciTeamFeatureSection(
        title = "CI Routing",
        badge = "CI",
        features = listOf(
            InciTeamFeature(
                id = "configuration-items",
                title = "CI",
                subtitle = "Supported systems",
                badge = "CI",
                tone = FeatureTone.Slate
            ),
            InciTeamFeature(
                id = "ci-user-mapping",
                title = "CI User Mapping",
                subtitle = "Routing ownership",
                badge = "CU",
                tone = FeatureTone.Slate
            )
        )
    ),
    InciTeamFeatureSection(
        title = "Operations",
        badge = "OP",
        features = listOf(
            InciTeamFeature(
                id = "summary",
                title = "Summary",
                subtitle = "Operational health",
                badge = "SU",
                tone = FeatureTone.Blue
            ),
            InciTeamFeature(
                id = "logs",
                title = "Logs",
                subtitle = "Assignment timeline",
                badge = "LG",
                tone = FeatureTone.Blue
            ),
            InciTeamFeature(
                id = "diagnostics",
                title = "Diagnostics",
                subtitle = "Routing checks",
                badge = "DX",
                tone = FeatureTone.Purple
            )
        )
    ),
    InciTeamFeatureSection(
        title = "Access",
        badge = "AC",
        features = listOf(
            InciTeamFeature(
                id = "account",
                title = "Account",
                subtitle = "Profile and deletion",
                badge = "ME",
                tone = FeatureTone.Red
            ),
            InciTeamFeature(
                id = "user-access",
                title = "User Access",
                subtitle = "Roles and teams",
                badge = "UA",
                tone = FeatureTone.Red
            )
        )
    )
)
